src/views/Register.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+    <layout :callback="callback">
+        <div class="registerBox">
+            <md-card class="registerCard">
+                <md-card-media-actions>
+                    <md-card-media>
+                        <img src="@/assets/images/garnbarn-logo.png" class="garnbarnLogo" />
+                    </md-card-media>
+
+                    <md-card-actions>
+                        <h3>Register to GarnBarn</h3>
+                        <div>
+                            <form @submit.prevent="register">
+                                <div class="user">
+                                    <label for="username">Username:</label>
+                                    <input id="username" v-model="username" type="text" />
+                                </div>
+                                <div class="user">
+                                    <label for="username">Email:</label>
+                                    <input id="username" v-model="email" type="text" />
+                                </div>
+                                <div class="pass">
+                                    <label for="password">Password:</label>
+                                    <input id="password" v-model="password" type="password" />
+                                </div>
+                                <div class="con-pass">
+                                    <label for="confirmPassword">Confirm Password:</label>
+                                    <input id="confirmPassword" v-model="confirmPassword" type="password" />
+                                </div>
+                                <div>
+                                    <button type="submit">Register</button>
+                                </div>
+                            </form>
+                        </div>
+                    </md-card-actions>
+                </md-card-media-actions>
+            </md-card>
+        </div>
+    </layout>
+</template>
+
+<script lang="ts">
+import firebase from "firebase";
+import DialogBox from "@/components/DialogBox/DialogBox";
+import { Vue, Component } from 'vue-property-decorator';
+import GarnBarnApi from "@/services/GarnBarnApi/GarnBarnApi";
+import GarnBarnApiConfig from "@/GarnBarnApiConfig.json";
+import axios from 'axios';
+import sha1 from 'sha1';
+
+@Component
+export default class Register extends Vue {
+    user: firebase.User | null = null;
+    username = '';
+    password = '';
+    confirmPassword = '';
+    email = ''
+
+    checkCompromisedPassword(): boolean {
+        const hashedPassword: string = sha1(this.password);
+        axios
+        .post(`${GarnBarnApiConfig.apiPrefix}/api/v1/account/compromised`, { "hashedPassword" : hashedPassword})
+        .then(response => {
+          return true
+        })
+        .catch(error => {
+          return false
+        });
+        return false
+    }
```

Comment:
API will return 200 if the password is NOT compromised.
But it will return 307 if the password IS compromised